[윤정환] sprint1

khuyjh opened the pull request:
## 요구사항

### 기본

- [x] 랜딩 페이지의 url path는 루트('/')로 설정합니다.
- [x] title은 "판다마켓"로 설정합니다.
- [x] 화면의 너비가 1920px 이상이면 하늘색 배경색은 너비를 꽉 채우도록 채워지고, 내부 요소들의 위치는 고정되고, 여백만 커지도록 합니다.
- [x] 화면의 너비가 1920px 보다 작아질 때, "판다마켓" 로고의 왼쪽 여백 200px"로그인" 버튼의 오른쪽 여백 200px이 유지되고, 화면의 너비가 작아질수록 두 요소간 거리가 가까워지도록 설정합니다
- [x] 클릭으로 기능이 동작해야 하는 경우, 사용자가 클릭할 수 있는 요소임을 알 수 있도록 CSS 속성 cursor: pointer 로 설정합니다.
- [x] "판다마켓" 클릭 시 루트 페이지('/')로 이동합니다.
- [x] '로그인'버튼 클릭 시 로그인 페이지('/login')로 이동합니다 (빈 페이지)
- [x] "구경하러가기"버튼 클릭 시('/items')로 이동합니다.(빈 페이지)
- [x] "Privacy Policy", "FAQ"는 클릭 시 각각 Privacy 페이지('/privacy'), FAQ 페이지('/faq')로 이동합니다.(모두 빈 페이지)
- [x] 페이스북, 트위터, 유튜브, 인스타그램 아이콘을 클릭 시 각각의 홈페이지로 새로운 창이 열리면서 이동합니다.

### 심화

- [ ] 사용자의 브라우저가 크고 작아짐에 따라 페이지의 요소간 간격, 요소의 크기, font-size 등 모든 크기와 관련된 값이 크고 작아지도록 설정해 보세요.(설정값은 자유입니다)

## 주요 변경사항

-
-

## 스크린샷

![스크린샷 2025-04-08 192504](https://github.com/user-attachments/assets/822b0e5e-1f09-4586-98a3-6df47912cf78)
![스크린샷 2025-04-08 192518](https://github.com/user-attachments/assets/b5413eee-2b86-4a4f-934f-fefade1064af)
![스크린샷 2025-04-08 192526](https://github.com/user-attachments/assets/46b2361d-2409-47cb-97a2-136b8afc2a2e)
![스크린샷 2025-04-08 192534](https://github.com/user-attachments/assets/b7d4e40f-b8c6-4223-a8c0-fc4fd6007569)
![스크린샷 2025-04-08 192543](https://github.com/user-attachments/assets/645e1a65-6542-4fe1-982b-5f9c9650324a)

## 멘토에게

-white-space: nowrap을 쓰게 되면 브라우저 크기를 줄여도 텍스트가 유지되어 정해진 컨테이너를 뚫고 나가게 되는데 어떻게 조절하면 좋을지 궁금합니다.
-flex속성의 컨테이너 너비를 브라우저 크기에 맞게 조절하기 위해서 너비를max-width로 주고 내부의 아이템을 flex:1 1 auto로 설정해두었는데, 최대 너비에서는 max-width로 설정한 최댓값보다 width값이 작게 나오고, 브라우저 크기를 줄였을 때는 img아이템의 크기가 줄어들지 않는 문제가 발생했습니다.

khuyjh committed:
mission1 first commit

khuyjh committed:
refactor(mentor): var를 let으로 수정

khuyjh committed:
refactor(mentor): var를 let으로 수정

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+<!DOCTYPE html>
+<html lang="en">
```
```suggestion
<html lang="ko">
```

언어 설정을 "ko"로 변경하시면, 한국어 사용자에 맞춰 스크린리더가 사용될수있고 검색엔진에서도 페이지의 언어를 한국어 콘텐츠로 인식하게끔 만들어줄 수 있습니다. 웹 접근성을 생각했을때 스크린리더를 사용한다면 lang="en" 상태에서는 한국어 콘텐츠를 영어 발음으로 읽을 가능성이 있기때문에, 한국어 사용자를 대상으로한다면 "ko"로 변경하시는게 옳습니다 :)

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+</head>
+<body>
+    <header>
+        <div class = "container_header">
+          <div class = "box_logo-and-url">
```
여러가지 네이밍 케이스가 혼재되어있네요 :)

[CSS 네이밍 컨벤션](https://velog.io/@krkorklo58/TIL-CSS-%EB%84%A4%EC%9D%B4%EB%B0%8D-%EC%BB%A8%EB%B2%A4%EC%85%98)
[자주 쓰는 네이밍케이스 정리](https://xionwcfm.tistory.com/459)

두개 참고해보시고, 일관적인 네이밍 규칙을 적용해보시면 좋을 것 같습니다!

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+        <div class = "container_header">
+          <div class = "box_logo-and-url">
+            <img src = "img/panda face.png" alt = "판다마켓 로고" class = "logo_header apply-cursor_pointer">
+            <a href = "/" class = "url_home">
```
접근성을 고려해 `<a href="/" aria-label="홈으로 이동">` 과 같이 aria-label 혹은 role을 추가해볼까요?

그리고 속성과 속성값사이의 띄어쓰기는 없애주세요!
포맷팅 신경써봅시다 :)

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+        </div>
+    </header>
+    <main>
+      <section class = "move-to-market">
```
전체적으로 클래스이름이 너무 구체적이고, 길어서 직관적으로 보이지않아요.
더 모듈화되고 재사용 가능한 패턴으로 조금 바꿔볼필요가 있겠네요 :)

예시를 들어 가장 많이 사용되는 CSS 네이밍 컨벤션인 BEM 기반 네이밍으로 바꿔보자면:

### 1. 블록(Block) 단위로 구조화
- `header`, `hero`, `feature`, `footer` 등 주요 섹션을 블록으로 정의
- 각 블록은 독립적인 컴포넌트로 취급

### 2. 요소(Element) 명명 규칙 적용
- 더블 언더스코어(__)를 사용하여 요소를 표시
예: `header__container`, `feature__title`, `footer__link`

### 3. 수정자(Modifier) 사용
- 더블 하이픈(--)을 사용하여 수정자를 표시
예: `feature--hot`, `feature--right`, `btn--primary`

참고해보시고 일괄적인 클래스이름 변경도 고려해보세요!

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+        </div>
+      </section>
+
+      <section class = "container_main first_container_main">
```
클래스이름을 지을때는 UI의 색상, 위치, 순서등이 포함되어있으면 디자인 변경 시 대응하기도 어려워지고 유지보수에 좋지 않아요. 이런 정보들을 포함하기보다는 항상 엘리먼트의 역할을 명확하게 나타낼수있는 네이밍을 써볼까요? :)

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+            믿을 수 있는<br>
+            판다마켓 중고 거래
+          </h2>
+          <img src = "img/Img_home_bottom.png" alt = "장바구니를 메고있는 판다 두 마리가 거래를 마치고 헤어지며 별점과 후기를 남기는 이미지" class = "img_bottom_main">
```
alt 텍스트 잘 사용해주셨네요 👍

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+    </main>
+
+    <footer>
+      <div class = "container_footer">
```
불필요한 div 중첩입니다 :)

addiescode-sj commented on index.html:
```diff
@@ -0,0 +1,123 @@
+          </a>
+        </div>
+        <div class = "icons_footer">
+          <a href = "https://www.facebook.com/?locale=ko_KR" target="_blank">
```
외부 링크의 경우, 보안을 위해 rel="noopener noreferrer"를 추가해주세요! :)

noopener: 새 탭에서 열린 페이지가 원본 페이지에 접근하는 것을 방지
noreferrer: 리퍼러 정보가 새 페이지로 전달되는 것을 방지

addiescode-sj commented on style.css:
```diff
@@ -0,0 +1,223 @@
+    src: url('fonts/ROKAF\ Sans\ Bold.ttf') format('truetype');
+}
+
+* {
+    box-sizing: border-box;
+    font-family: 'Pretendard-Regular';
+    white-space: nowrap; /*브라우저 줄였을 때 텍스트 줄바뀜 방지*/
+}
```
용도에 따라 CSS를 나눠서 관리해볼까요?

- reset.css: CSS 초기화
- common.css: 전역 스타일

### 이렇게 용도에 따라 파일을 나눠 관리하는건 어떤 이점을 가져다줄수있는지 한번 더 생각해볼수있어요.

- 역할 분리가 되어 관리가 쉬움: 각 파일의 목적이 명확해서 수정이나 유지보수가 쉬워질거예요.
- 사용성과 확장성 증가: 여러 페이지 혹은 프로젝트에서 거의 동일하게 재사용이 가능해요.
- 스타일 우선순위가 더 예측 가능해짐: reset.css → common.css → 페이지나 컴포넌트별 스타일 순으로 로딩하면 우선순위가 명확해지고 스타일 충돌을 줄일 수 있어요.

addiescode-sj commented on style.css:
```diff
@@ -0,0 +1,223 @@
+}
+
+.url_home {
+    color: #3692ff;
```
자주 사용되는 값들은 css 변수로 만들어서 재사용하면 코드 중복이 줄어들고, 유지보수에도 유리하겠죠?

```
:root {
  /* Colors */
  --color-primary: #3692ff;
  --color-primary-dark: #1967d6;
  --color-primary-darker: #1251aa;
  --color-disabled: #9ca3af;
  --color-background-light: #fcfcfc;
  --color-background-blue: #cfe5ff;
  --color-text-dark: #111827;
  --color-text-light: #e5e7eb;
  --color-text-gray: #9ca3af;
  --color-white: #ffffff;

  /* Typography */
  --font-primary: "Pretendard-Regular", sans-serif;
  --font-secondary: "ROKAF SANS", sans-serif;

  /* Spacing */
  --spacing-xs: 5px;
  --spacing-sm: 10px;
  --spacing-md: 16px;
  --spacing-lg: 24px;
  --spacing-xl: 32px;
  --spacing-xxl: 64px;

  /* Layout */
  --container-width: 1520px;
  --container-min-width: 820px;
  --content-width: 1110px;
  --feature-width: 988px;
}
```

addiescode-sj reviewed (approved):
수고하셨습니다! 
전체적으로 조금만 다듬으면 멋진 프로젝트 나올것같아요~!

## 주요 리뷰 포인트
- 클래스이름 변경
- 접근성 향상
- 코드 중복 제거
- 포맷팅

addiescode-sj commented:
> ## 멘토에게
> - white-space: nowrap을 쓰게 되면 브라우저 크기를 줄여도 텍스트가 유지되어 정해진 컨테이너를 뚫고 나가게 되는데 어떻게 조절하면 좋을지 궁금합니다. 
> - flex속성의 컨테이너 너비를 브라우저 크기에 맞게 조절하기 위해서 너비를max-width로 주고 내부의 아이템을 flex:1 1 auto로 설정해두었는데, 최대 너비에서는 max-width로 설정한 최댓값보다 width값이 작게 나오고, 브라우저 크기를 줄였을 때는 img아이템의 크기가 줄어들지 않는 문제가 발생했습니다.

- white-space: nowrap을 썼을때 브라우저 크기를 어느정도로 줄이면 컨테이너를 뚫고 나가게되는지 사이즈 측정해보시고, 반응형을 고려해 미디어쿼리로 속성을 바꿔주시거나 `word-break: keep-all` 과 같은 음절단위 자동 줄바꿈 속성을 쓰시는것도 좋은 방법입니다 :) [mdn 참고](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break) 보통 nowrap은 문장에 적용하지않고 줄바꿈을 하지않아야하는 버튼 텍스트 등에 적용하는 편이예요. 어떤 화면에서든 유연한 텍스트 영역을 가질수있도록 고려해주세요!

- flex 속성과 max-width 속성이 충돌해서 그렇습니다. 개발자 도구에서 확인해보시면 워닝이 표시된다던지 글자가 흐려져있거나 중간선이 쳐진걸 확인하실수있을거예요. 해결 방안은, flex container에는 `width: 100%` 와 함께 `max-width` 를 추가하시고 내부 아이템의 경우 `flex: 1 1 0; max-width: 100%`를 설정해보세요 :) 이렇게 하면 내부 아이템이 올바로 컨테이너 크기에 맞춰질거예요.

khuyjh committed:
common.css reset.css파일 분리

khuyjh committed:
html lang ="ko"로 변경
속성과 속성값 사이 띄어쓰기 제거
class 네이밍 수정
a태그 요소에 aria-label과 role 추가
footer 섹션 불필요한 div삭제
외부 링크에 rel="noopener noreferrer" 추가
root가상 클래스를 통해 자주쓰는 색상, 폰트 변수화
버튼 텍스트에 white-space:nowrap, 그 외 모든 문장에 word-break: keep-all
적용
flex 컨테이너에 width:100% max-width 동시적용, 아이템에 flex 1 1 0,
max-width:100% 동시적용
